avro-converter/src/main/java/io/confluent/connect/avro/AvroData.java

```diff
@@ -1013,6 +1018,26 @@ private void addAvroRecordField(
     fields.add(field);
   }
 
+  private static Object fromLogical(Schema schema, Object value) {
+    if (schema != null && schema.name() != null) {
+      LogicalTypeConverter logicalConverter = TO_AVRO_LOGICAL_CONVERTERS.get(schema.name());
+      if (logicalConverter != null && value != null) {
+        return logicalConverter.convert(schema, value);
+      }
+    }
+    return value;
+  }
```

Comment:
This was factored out of the `defaultValueFromConnect(...)` method so it can be used in multiple places.